src/memory.rs

```diff
@@ -55,8 +55,12 @@ impl Pointer {
     pub fn points_to_zst(&self) -> bool {
         self.alloc_id == ZST_ALLOC_ID
     }
-    pub fn to_int(&self) -> usize {
-        self.offset
+    pub fn to_int<'tcx>(&self) -> EvalResult<'tcx, usize> {
```

Comment:
Most uses don't have `.expect`. Those that do need to be fixed, because the following example would end up triggering this assertion, due to using an abstract pointer as an index. It should not bug out, but report a regular miri error.

```rust
let arr = [5, 6, 7];
let something = 42;
let x = arr[&something as *const i8 as usize];
```